.github/workflows/prepare-release.yml

```diff
@@ -19,8 +19,35 @@ on:
         required: true
         description: Current version (beta, like 0.69.1). Don't include `v`.
 jobs:
+  #validate-version format
+  validate-versions:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Validate version format
+      run: |
+        validate_version() {
+          local regex_pattern_beta='^[0-9]+\.[1-9][0-9]*\.[1-9][0-9]*$'
```

Comment:
This prevents setting the patch number to `0`. This should be allowed.

Reply:
for beta - major can be 0-9,  minor can be 1-9, patch can be 0-9
for stable -  major can be 1-9, minor can be 0-9 and patch can be 0-9 
and RC version shall be optional.

right?

Reply:
Looking at this [comment](https://github.com/open-telemetry/opentelemetry-collector/pull/9409#discussion_r1471625991) from Tyler:

> Stable should start with a one or higher and include a Major.Minor.Patch version, which no RC candidates. Beta should be the same but the Major version can be a zero.

> for beta - major can be 0-9, minor can be 1-9, patch can be 0-9

Just do 0-9, 0-9, 0-9

> for stable - major can be 1-9, minor can be 0-9 and patch can be 0-9

Yes

> and RC version shall be optional.

We should not allow RC version at all, let's remove it.